quarkchain/evm/messages.py

```diff
@@ -366,6 +366,7 @@ def __init__(self, state, tx):
         self.reset_storage = state.reset_storage
         self.tx_origin = tx.sender if tx else b"\x00" * 20
         self.tx_gasprice = tx.gasprice if tx else 0
+        self.posw_disallow_list = state.posw_disallow_list
```

Comment:
nit:
following the same pattern here, we can add a `is_locked` function to state and then pass that function to VMExt.

Reply:
I think it's more of a functional approach, and mostly a coding style thing. personally I think passing the list to VM is rather more straightforward and easier to read (unless in the future we would have a more complicated logic on disallowing senders)